boards: dts: Fix dtc led warning in stm32 board dts files

erwango opened the pull request:
This change aims at fixing 'unit_address_vs_reg' warning in
STM32 based boards.
This warning pops up when a node name is made up with an address
(node_name@xx) but does not contain a reg property.
This case was encountered for led nodes for instance,
where a reg property has no meaning.
Fix this by changing node_name@xx to node_name_xx which removes the
 guilty '@xx' syntax but preserves node_name uniqueness.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ydamigos commented on boards/arm/nucleo_f207zg/nucleo_f207zg.dts:
```diff
@@ -19,15 +19,15 @@
 		compatible = "gpio-leds";
-		green_led_1: led@0 {
+		green_led_1: led_1 {
 			gpios = <&gpiob 0 GPIO_INT_ACTIVE_HIGH>;
```
According to the DT specification:
```
If the node has no reg property, the @unit-address must be omitted and the node-name alone
differentiates the node from other nodes at the same level in the tree.
```
The node-name here is `led_1` for all nodes.

galak replied:
@ydamigos is this resolved?

ydamigos commented on boards/arm/nucleo_f412zg/nucleo_f412zg.dts:
```diff
@@ -19,15 +19,15 @@
 	leds {
 		compatible = "gpio-leds";
-		green_led_1: led@0 {
+		green_led_1: led_1 {
```
The same here.

erwango replied:
Right. Script abuse is bad. Will fix. Thanks

ydamigos commented on boards/arm/nucleo_f413zh/nucleo_f413zh.dts:
```diff
@@ -19,15 +19,15 @@
 	leds {
 		compatible = "gpio-leds";
-		green_led_1: led@0 {
+		green_led_1: led_1 {
```
The same here.

ydamigos commented on boards/arm/nucleo_f429zi/nucleo_f429zi.dts:
```diff
@@ -20,15 +20,15 @@
 	leds {
 		compatible = "gpio-leds";
-		green_led_1: led@0 {
+		green_led_1: led_1 {
```
The same here.

ydamigos commented on boards/arm/stm32f072b_disco/stm32f072b_disco.dts:
```diff
@@ -19,19 +19,19 @@
 			label = "User LD3";
 		};
-		yellow_left_4: led@1 {
+		yellow_left_4: led_1 {
```
led_4?

ydamigos reviewed (changes requested):
If we omit the @unit-address, the node-name should differentiate the node from other nodes at the same level in the tree.

erwango committed:
boards: dts: Fix dtc warning in stm32 board dts files

This change aims at fixing 'unit_address_vs_reg' warning in
STM32 based boards.
This warning pops up when a node name is made up with an address
(node_name@xx) but does not contain a reg property.
This case was encountered for led nodes for instance,
where a reg property has no meaning.
Fix this by changing node_name@xx to node_name_xx which removes the
 guilty '@xx' syntax but preserves node_name uniqueness.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak reviewed (changes requested):
The node names, should not just convert @0,1,2 to _1,2,3.  The names should be more related to how the board/board docs describe the led.

So for example on 96b_carbon:

green_led_1: led@0 -> green_led_1: led_usr1
green_led_1: led@1 -> green_led_2: led_usr2
bt_blue_led: led@2 -> bt_blue_led: led_bt

Mani-Sadhasivam commented:
@galak Node name should describe the class of the device, which is just `led` in this case. So, I would vote for `led_n`. The device description like `usr` can then go into label.

ydamigos reviewed (changes requested):
You missed on file.

galak commented:
> You missed on file.

I went ahead and fixed that one.

erwango commented:
recheck
